### ISSUE TITLE
losses don't drop when training on custom data

### ISSUE THREAD

**Issue by GengyuanMax** [nianticlabs/monodepth2#247]
Hi,

when I train a new model on my custom dataset, the loss hardly drops through training procedure. I don't have valid depth groud truth so I cannot see whether the metrics are improved. But I think it is still weird.

My current assumptions are:
1. camera movement has a short baseline because I use an indoor dataset and extract one frame per 0.4 sencond
2. I used a training epoch of 30, which is not enough 

Could you give me any advice?

Thanks.

**Comment by mrharicot** [comment]
Hi,
It is hard to tell without looking at some training data.
Would you mind sharing some example images ? (if possible 3 temporal training frames)

**Comment by GengyuanMax** [comment]
> Hi,
> It is hard to tell without looking at some training data.
> Would you mind sharing some example images ? (if possible 3 temporal training frames)

Hi, 
I used the given pose between frames.

As shown, each triplet is a sample. (current frame, previous frame, next frame)

![image](https://user-images.githubusercontent.com/33713574/94174745-f871a100-fe95-11ea-85f4-b7f7309deeb8.png)
![image](https://user-images.githubusercontent.com/33713574/94174778-00c9dc00-fe96-11ea-92b3-d5c37d993c50.png)

The camera motion is very variant so the training result is not very ideal.

**Comment by blacksino** [comment]
> > Hi,
> > It is hard to tell without looking at some training data.
> > Would you mind sharing some example images ? (if possible 3 temporal training frames)
> 
> Hi,
> I used the given pose between frames.
> 
> As shown, each triplet is a sample. (current frame, previous frame, next frame)
> 
> ![image](https://user-images.githubusercontent.com/33713574/94174745-f871a100-fe95-11ea-85f4-b7f7309deeb8.png)
> ![image](https://user-images.githubusercontent.com/33713574/94174928-3a9ae280-fe96-11ea-8f5a-0566487b5144.png)
> 
> The camera motion is very variant so the training result is not very ideal.

你好，我也在复现monodepth2，我目前也遇到和你一样的情况，loss不下降，而且loss一开始就很低，我们可以一起讨论一下？

**Comment by GengyuanMax** [comment]
> > > Hi,
> > > It is hard to tell without looking at some training data.
> > > Would you mind sharing some example images ? (if possible 3 temporal training frames)
> > 
> > 
> > Hi,
> > I used the given pose between frames.
> > As shown, each triplet is a sample. (current frame, previous frame, next frame)
> > ![image](https://user-images.githubusercontent.com/33713574/94174745-f871a100-fe95-11ea-85f4-b7f7309deeb8.png)
> > ![image](https://user-images.githubusercontent.com/33713574/94174928-3a9ae280-fe96-11ea-8f5a-0566487b5144.png)
> > The camera motion is very variant so the training result is not very ideal.
> 
> 你好，我也在复现monodepth2，我目前也遇到和你一样的情况，loss不下降，而且loss一开始就很低，我们可以一起讨论一下？

Cool, how can i reach you? you could find my email in my page

**Comment by mrharicot** [comment]
Training self-supervised monocular depth models on indoors data with complex camera motion is a hard problem which is still an active research topic.
The pose prediction is quite difficult as well as the presence of a lot of textureless regions (walls, etc.).
My main recommendation would be to use lower resolution inputs to start with and to try and limit the camera motion to in-plane translation (camera moves up/down, or left/right, rather than forward, backward as this is a much harder case for triangulation).
Good luck!
